src/app/attendance/components/AttendanceFormBuilder.tsx

```diff
@@ -0,0 +1,270 @@
+    defaultValues: {
+      memberId: loginUser.role === MemberRole.Member ? loginUser.id : undefined,
+      startTime: "",
+      endTime: "",
+    },
+  });
+
+  const roleBaseMembers = useMemo(() => {
+    return members
+      .filter(
+        (member) =>
+          member.status === UserStatus.Active &&
+          (loginUser.role === MemberRole.Incharge ||
+            member.shift === loginUser.shift)
+      )
+      .map(({ id, name }) => ({
+        id,
+        name,
+      }));
+  }, []);
+
+  const handleUserSelect = (value: string) => {
+    form.setValue("memberId", Number(value), { shouldValidate: true });
+  };
+
+  const onSubmit = async (values: AttendanceFormValues) => {
+    console.log({ values });
+  };
+
+  return (
+    <Form {...form}>
+      <form
+        onSubmit={form.handleSubmit(onSubmit)}
+        className="max-w-lg mx-auto p-8 mt-10 bg-white shadow-md rounded-md space-y-6"
+      >
+        <h1 className="text-2xl font-bold text-center mb-6">Attendance Form</h1>
+
+        <FormField
+          control={form.control}
+          name="memberId"
+          render={({ field }) => (
+            <FormItem>
+              <FormLabel>User Name</FormLabel>
+              <FormControl>
+                <Select
+                  value={field.value?.toString() || ""}
+                  onValueChange={handleUserSelect}
+                  disabled={loginUser.role === "member"}
+                >
+                  <SelectTrigger
+                    className={`border rounded-md p-2 ${
+                      form.formState.errors.memberId
+                        ? "border-red-500"
+                        : "border-gray-300"
+                    }`}
+                  >
+                    <SelectValue placeholder="Select a user">
+                      {field.value
+                        ? roleBaseMembers.find((m) => m.id === field.value)
+                            ?.name
+                        : "Select a user"}
+                    </SelectValue>
+                  </SelectTrigger>
+                  <SelectContent>
+                    {roleBaseMembers.map((user) => (
+                      <SelectItem key={user.id} value={user.id.toString()}>
+                        {user.name}
+                      </SelectItem>
+                    ))}
+                  </SelectContent>
+                </Select>
+              </FormControl>
+              <FormMessage />
+            </FormItem>
+          )}
+        />
+
+        <FormField
+          control={form.control}
+          name="startTime"
+          render={({ field }) => (
+            <FormItem>
+              <FormLabel>Start Time</FormLabel>
+              <FormControl>
+                <Input
+                  type="time"
+                  {...field}
+                  onClick={(event) => event.currentTarget.showPicker()}
+                  className={`border rounded-md p-2 ${
+                    form.formState.errors.startTime
```

Comment:
I have update input according to error so we will fix out after merging of my PR